homeassistant/components/elv/switch.py

```diff
@@ -1,38 +1,29 @@
 """Support for PCA 301 smart switch."""
 import logging
 
-import voluptuous as vol
-
 from homeassistant.components.switch import (
-    SwitchDevice, PLATFORM_SCHEMA, ATTR_CURRENT_POWER_W)
-from homeassistant.const import (
-    CONF_NAME, CONF_DEVICE, EVENT_HOMEASSISTANT_STOP)
-import homeassistant.helpers.config_validation as cv
+    SwitchDevice, ATTR_CURRENT_POWER_W)
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from . import DOMAIN as ELV_DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_TOTAL_ENERGY_KWH = 'total_energy_kwh'
 
 DEFAULT_NAME = 'PCA 301'
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Required(CONF_DEVICE): cv.string
-})
-
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the PCA switch platform."""
     import pypca
```

Comment:
Add a guard clause here that checks if discovery_info is None and return if so.

Please move the 3rd party imports to the top of the module.